runtime/drivers/druid/olap.go

```diff
@@ -196,7 +196,15 @@ func (i informationSchema) All(ctx context.Context) ([]*drivers.Table, error) {
 }
 
 func (i informationSchema) Lookup(ctx context.Context, db, schema, name string) (*drivers.Table, error) {
-	q := `
+	// ensure Coordinator is ready
+	q := "SELECT * FROM sys.segments LIMIT 1"
+	rows, err := i.c.db.QueryxContext(ctx, q, name)
+	if err != nil {
+		return nil, err
+	}
+	rows.Close()
```

Comment:
It's only on Lookup query and it's negligible compared to the whole amount of other requests.
The issues is that the request
```
		SELECT
			T.TABLE_SCHEMA AS SCHEMA,
			T.TABLE_NAME AS NAME,
			T.TABLE_TYPE AS TABLE_TYPE, 
			C.COLUMN_NAME AS COLUMN_NAME,
			C.DATA_TYPE AS COLUMN_TYPE,
			C.IS_NULLABLE = 'YES' AS IS_NULLABLE
		FROM INFORMATION_SCHEMA.TABLES T 
		JOIN INFORMATION_SCHEMA.COLUMNS C ON T.TABLE_SCHEMA = C.TABLE_SCHEMA AND T.TABLE_NAME = C.TABLE_NAME
		WHERE T.TABLE_SCHEMA = 'druid' AND T.TABLE_NAME = ?
		ORDER BY SCHEMA, NAME, TABLE_TYPE, C.ORDINAL_POSITION
```
returns false-negative if the Coordinator is being restarted. Retrier is a more abstract class and it doesn't check if SQL tries to retreive dynamic schema and the will be no error from Druid Router (because if dynamic schema is empty - it's considered OK by the cluster).

Reply:
added a description.